ansible/roles/runner/tasks/custom_vagrant_catalog.yml

```diff
@@ -1,18 +1,27 @@
 ---
 # Configure host to fetch boxes from custom catalog (prci-automation)
-- name: Add prci-automation entry to /etc/hosts
-  lineinfile:
-    path: /etc/hosts
-    regexp: "prci-automation$"
-    line: "{{ custom_vagrant_catalog }} prci-automation"
-
-- name: Set "VAGRANT_SERVER_URL" Environment variable
+- name: Reset "VAGRANT_SERVER_URL" Environment variable
   ini_file:
     path: /etc/systemd/system/prci.service
     section: Service
     option: Environment
-    value: "VAGRANT_SERVER_URL='http://prci-automation'"
-    mode: 0700
+    state: absent
+
+- block:
+    - name: Add prci-automation entry to /etc/hosts
+      lineinfile:
+        path: /etc/hosts
+        regexp: "prci-automation$"
+        line: "{{ custom_vagrant_catalog }} prci-automation"
+
+    - name: Set "VAGRANT_SERVER_URL" Environment variable
+      ini_file:
+        path: /etc/systemd/system/prci.service
+        section: Service
+        option: Environment
+        value: "VAGRANT_SERVER_URL='http://prci-automation'"
+        mode: 0700
+  when: custom_vagrant_catalog is defined and custom_vagrant_catalog != 'False'
```

Comment:
The variable has `default: false`, so result of the condition `custom_vagrant_catalog != 'False'` will be always true.

Reply:
At some point, Ansible cast that boolean to string, that's why I'm comparing with `"False"`, I believe because that's `prompt`'s behaviour, since it's expecting an input from the user.

```python
ansible-playbook test.yaml                                                                                                     

Custom vagrant catalog host (ip address) (Press enter to ignore) [False]: 

PLAY [prepare development test runners] ****************************************************************************************************

TASK [debug] *******************************************************************************************************************************
ok: [localhost] => {
    "msg": "Value False"
}

TASK [check we're not deploying against FreeIPA repo] **************************************************************************************
ok: [localhost] => {
    "custom_vagrant_catalog": false,
    "failed_when_result": false
}

PLAY RECAP *********************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```
```python
ansible-playbook test.yaml                                                                                                     

Custom vagrant catalog host (ip address) (Press enter to ignore) [False]: foo

PLAY [prepare development test runners] ****************************************************************************************************

TASK [debug] *******************************************************************************************************************************
ok: [localhost] => {
    "msg": "Value foo"
}

TASK [check we're not deploying against FreeIPA repo] **************************************************************************************
fatal: [localhost]: FAILED! => {
    "custom_vagrant_catalog": "foo",
    "failed_when_result": true
}

PLAY RECAP *********************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
```